File: Week_02/question49/Solution.py
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        dp, a, b, c = [1] * n, 0, 0, 0
        for i in range(1, n):
            da = dp[a] * 2
            db = dp[b] * 3
            dc = dp[c] * 5
            dp[i] = min(da, db, dc)
            if dp[i] == da:
                a += 1
            if dp[i] == db:
                b += 1
            if dp[i] == dc:
                c += 1
        return dp[n - 1]

File: Week_02/question49/Solution.java
package com.leetcode.question49;


public class Solution {
    public int nthUglyNumber(int n) {
        int[] dp = new int[n];
        dp[0] = 1;
        int a = 0;
        int b = 0;
        int c = 0;
        for(int i=1;i<n;i++){
            int da = dp[a]*2;
            int db = dp[b]*3;
            int dc = dp[c]*5;
            dp[i] = Math.min(Math.min(da,db),dc);
            if(dp[i]==da) a++;
            if(dp[i]==db) b++;
            if(dp[i]==dc) c++;
        }
        return dp[n-1];
    }
}
